docs/src/logging.md

```diff
@@ -0,0 +1,145 @@
+# Logging
+
+## Overview
+
+Logging is a crucial tool for debugging, monitoring, and understanding the behavior of your applications. ClimaComms extends Julia's built-in logging functionality (provided by `Logging.jl`) to work seamlessly in distributed computing environments, particularly with MPI.
+
+### Julia's Logging System
+
+Julia's standard library provides a flexible logging system through `Logging.jl`. Key concepts include:
+
+- **Logger**: An object that handles log messages, determining how they are formatted and where they are sent
+- **Log Level**: Indicates the severity/importance of a message
+- **Log Record**: Contains the message and associated metadata (level, module, line number, etc.)
+
+The default logger in Julia (as of v1.11) is `Logging.ConsoleLogger()`, which prints messages to the console. You can check your current logger using:
+
+```julia
+using Logging; current_logger()
+```
+
+## ClimaComms Logging Features
+
+
+ClimaComms builds upon Julia's logging system by providing specialized loggers for distributed computing:
+
+### MPIFileLogger
+
+`MPIFileLogger(context, log_dir)` separates logs by MPI rank into different files. By default, the root rank is logged to `stdout` as well. This can be disabled by setting the keyword argument `log_stdout = false`.
+
+```julia
+using ClimaComms, Logging
+ctx = ClimaComms.context()
+logger = ClimaComms.MPIFileLogger(ctx, "logs")
+
+with_logger(logger) do
+    @warn "Memory usage high"  # Written to rank-specific log file
+end
+```
+
```

Comment:
We should add a line describing the expected output of this example. This would emphasize where the files are saved and with which naming convention.